src/connectors/impls/kafka/consumer.rs

```diff
@@ -329,14 +386,48 @@ impl KafkaConsumerSource {
             origin_uri,
             topics,
             topic_resolver,
-            transactional: !auto_commit,
+            transactional: !config.auto_commit,
             retry_failed_events,
             seek_timeout,
             source_tx,
             source_rx,
             consumer: None,
             consumer_task: None,
             metrics_rx: None,
+            last_rebalance_ts: Arc::new(AtomicU64::new(0)),
+            cached_assignment: None,
+        }
+    }
+
+    /// gets the current assignment from the cache or fetches it from the group coordinator
```

Comment:
I might not have the full context here, but from what I understand, this really sits wrong with me, it feels like we're trying to outsmart kafka/librkafka and make it do something that it's not meant to do.

The problem this is trying to fix, from what I can tell, is a quick succession of CB-open CB-close events. Would be changing the CB system to prevent these floppy changes a option, it seems like behaviour like that is potentially harmful to all connectors and not just this one.

Reply:
Yep - we could separate this out so that any connector can use a configurable failure detector which would make the fix generally available to all connectors.